exporter/metric/metric_test.go

```diff
@@ -422,3 +429,76 @@ func TestTimeIntervalPassthru(t *testing.T) {
 	assert.Equal(t, start, tm)
 	assert.Equal(t, end, tm.Add(time.Second))
 }
+
+type mock struct {
+	monitoringpb.UnimplementedMetricServiceServer
+	createTimeSeries       func(ctx context.Context, req *monitoringpb.CreateTimeSeriesRequest) (*empty.Empty, error)
+	createMetricDescriptor func(ctx context.Context, req *monitoringpb.CreateMetricDescriptorRequest) (*googlemetricpb.MetricDescriptor, error)
+}
+
+func (m *mock) CreateTimeSeries(ctx context.Context, req *monitoringpb.CreateTimeSeriesRequest) (*empty.Empty, error) {
+	return m.createTimeSeries(ctx, req)
+}
+
+func (m *mock) CreateMetricDescriptor(ctx context.Context, req *monitoringpb.CreateMetricDescriptorRequest) (*googlemetricpb.MetricDescriptor, error) {
+	return m.createMetricDescriptor(ctx, req)
+}
+
+func TestExportMetricsWithUserAgent(t *testing.T) {
+	server := grpc.NewServer()
+	t.Cleanup(server.Stop)
+
+	// Channel to shove user agent strings into
+	ch := make(chan []string, 2)
+
+	m := mock{
+		createTimeSeries: func(ctx context.Context, r *monitoringpb.CreateTimeSeriesRequest) (*empty.Empty, error) {
+			md, _ := metadata.FromIncomingContext(ctx)
+			ch <- md.Get("User-Agent")
+			return &emptypb.Empty{}, nil
+		},
+		createMetricDescriptor: func(ctx context.Context, req *monitoringpb.CreateMetricDescriptorRequest) (*googlemetricpb.MetricDescriptor, error) {
+			md, _ := metadata.FromIncomingContext(ctx)
+			ch <- md.Get("User-Agent")
+			return req.MetricDescriptor, nil
+		},
+	}
+	monitoringpb.RegisterMetricServiceServer(server, &m)
+
+	lis, err := net.Listen("tcp", "127.0.0.1:0")
+	require.NoError(t, err)
+	go server.Serve(lis)
+
+	clientOpts := []option.ClientOption{
+		option.WithEndpoint(lis.Addr().String()),
+		option.WithoutAuthentication(),
+		option.WithGRPCDialOption(grpc.WithInsecure()),
+	}
+	res := &resource.Resource{}
+	cps := metrictest.NewCheckpointSet(res)
+	ctx := context.Background()
+	desc := metric.NewDescriptor("testing", metric.ValueRecorderInstrumentKind, number.Float64Kind)
+
+	lvagg := &lastvalue.New(1)[0]
+	aggtest.CheckedUpdate(t, lvagg, number.NewFloat64Number(12.34), &desc)
+	lvagg.Update(ctx, number.NewFloat64Number(12.34), &desc)
+	cps.Add(&desc, lvagg, label.String("a", "A"), label.String("b", "B"))
+
+	opts := []Option{
+		WithProjectID("PROJECT_ID_NOT_REAL"),
+		WithMonitoringClientOptions(clientOpts...),
+		WithMetricDescriptorTypeFormatter(formatter),
+	}
+
+	exporter, err := NewRawExporter(opts...)
+	if err != nil {
+		t.Errorf("Error occurred when creating exporter: %v", err)
+	}
+
+	err = exporter.Export(ctx, cps)
+	assert.NoError(t, err)
+
+	// Now check for user agent string in the buffer from one call.
```

Comment:
Is this missing a check for the other call?

Reply:
Do you think there's value in checking both?   What're the odds you misconfigure JUST ONE grpc method user agent without misconfiguring both?